### PR TITLE
Internal: Allow extending Add New modal from Pro plugin [ED-7730]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

**Review by TzviRabinovitch — DISMISSED** [review]

File: assets/dev/js/admin/new-template/layout.js [hunk 1]
```diff
@@ -20,6 +20,110 @@ module.exports = elementorModules.common.views.modal.Layout.extend( {
...
+		this.elements.submitButton.classList.add( 'e-hidden' );
+	},
+
+	replaceLockLinkPlaceholders( link ) {
+		// %1$s => utm_source
+		// %2$s => utm_medium
+
+		return link
+			.replace( /%1\$s/g, 'wp-add-new' )
+			.replace( /%2\$s/g, 'wp-dash' );
```
Comment (matipojo) [inline assets/dev/js/admin/new-template/layout.js:126]: {placeholder}

File: includes/admin-templates/new-template.php [hunk 2]
```diff
@@ -65,11 +70,23 @@
...
+
 				<select id="elementor-new-template__form__template-type" class="elementor-form-field__select" name="template_type" required>
 					<option value=""><?php echo esc_html__( 'Select', 'elementor' ); ?>...</option>
 					<?php
 					foreach ( $types as $value => $type_title ) {
-						printf( '<option value="%1$s" %2$s>%3$s</option>', esc_attr( $value ), selected( $selected, $value, false ), esc_html( $type_title ) );
+						printf(
+							'<option value="%1$s" data-lock=\'%2$s\' %3$s>%4$s</option>',
+							esc_attr( $value ),
+							wp_json_encode( $lock_configs[ $value ] ),
```
Comment (matipojo) [inline includes/admin-templates/new-template.php:86]:
```suggestion
							esc_attr( wp_json_encode( $lock_configs[ $value ] ) ),
```

**Review by matipojo — CHANGES_REQUESTED** [review]
Please cover with tests

File: includes/admin-templates/new-template.php (hunk 2 above)
Comment (StyleShit) [reply includes/admin-templates/new-template.php:86]: PHPCS doesn't require it

File: assets/dev/js/admin/new-template/layout.js (hunk 1 above)
Comment (StyleShit) [reply assets/dev/js/admin/new-template/layout.js:126]: Done

**Pull request merged**